### DISCUSSION TITLE
v0.53.0

### DISCUSSION THREAD

**Discussion by chen-keinan** [aquasecurity/trivy#7061] [Announcements]
# 📑 Table of Contents

- 💔 Breaking Change 💔
  -  🌵 Cache Management Flags Removed 🌵 
  - ☁️Trivy AWS is now available via a plugin ☁
  - 🫓 Trivy misconfiguration schema flattened 🚜
  - ➗ Trivy now scans `composer.lock` files only in `fs` and `repo` modes. 🧩
  -  Kuberetes compliance changes
- 🚀 What's new? 🚀
  - 🐍 Support for License Detection for `environment.yml` Files (Conda) 📜
  - 🔎 Support for determining `Dart` dependency version from SDK constraint 🔍
  -  🗃️ Support for `maven-metadata.xml` for `maven snapshot repositories` 📥
  - 🎊 Support for `installed.json` PHP files ✨
  - 🌀 Trivy migrated to `CycloneDX v1.6` 🌪️
  - 🧠 Memory Cache Backend ⚡
  - 🥡 Node collector commands ⛏️
  - 🧾 New compliance specs ☑️
  - 🫔 Selector support for all providers 🍏
  - 🐳 New resources support in misconfiguration scanning 🎍
  - 📦 Plugin GitHub Archives Support 🗂️
- 👷‍♂️ Notable Fixes 🛠️

#  💔 Breaking Changes 💔

## 🌵 Cache Management Flags Removed 🌵 
As announced [here](https://github.com/aquasecurity/trivy/discussions/7010), the following cache management flags will be removed:

- --reset
- --clear-cache
- --reset-checks-bundle

These flags will be replaced by the new `trivy clean` command.
See [here](https://github.com/aquasecurity/trivy/discussions/7010) for more details.

## ☁️Trivy AWS is now available via a plugin ☁

As [announced before](https://github.com/aquasecurity/trivy/discussions/6818), Trivy's AWS scanning is now offered as a separate plugin. It is no longer shipped with Trivy by default, but can be easily installed as such:

```shell
trivy plugin install github.com/aquasecurity/trivy-aws
```

## 🫓 Trivy misconfiguration schema flattened 🚜

As [announced before](https://github.com/aquasecurity/trivy/discussions/6883) Trivy's internal cloud schema has been improved to remove potential recursions. This is relevant only for rule authors, and does not effect users.

## Kuberetes compliance changes

Trivy now defines a naming convention for compliance reports in the form of: `{platform}-{type}-{version}`. Following are the renamed specs:

- `k8s-cis` -> `k8s-cis-1.23`
- `k8s-nsa` -> `k8s-nsa-1.0`
- `k8s-pss-baseline` -> `k8s-pss-baseline-0.1`
- `k8s-pss-restricted` -> `k8s-pss-restricted-0.1`
- `docker-cis` -> `docker-cis-1.6.0`

## ➗ Trivy now scans `composer.lock` files only in `fs` and `repo` modes. 🧩
We have added support for `installed.json` files.
So we separated the PHP files:
- `composer.lock` for `fs` and `repo` modes.
- `installed.json` for `image` and `rootfs` modes.

# 🚀 What's new? 🚀

## 🐍 Support for License Detection for `environment.yml` Files (Conda) 📜
Trivy now supports license detection for `environment.yml` files by parsing `<package.json>` files in [prefix](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#specifying-a-location-for-an-environment) directory. 

```bash
environment.yaml (license)

Total: 1 (UNKNOWN: 0, LOW: 1, MEDIUM: 0, HIGH: 0, CRITICAL: 0)

┌───────────────┬──────────────┬────────────────┬──────────┐
│    Package    │   License    │ Classification │ Severity │
├───────────────┼──────────────┼────────────────┼──────────┤
│ _openmp_mutex │ BSD-3-Clause │ Notice         │ LOW      │
└───────────────┴──────────────┴────────────────┴──────────┘
```
Read more [here](https://aquasecurity.github.io/trivy/v0.53/docs/coverage/os/conda/#license_1).

## 🔎 Support for determining `Dart` dependency version from SDK constraint 🔍
For `Dart` dependencies that use SDK as version, Trivy now uses the first version of the SDK constraint.

For example in this case the version of `flutter_test` will be `3.3.0`:
```
flutter_test:
    dependency: "direct dev"
    description: flutter
    source: sdk
    version: "0.0.0"
sdks:
  dart: ">=2.18.0 <3.0.0"
  flutter: "^3.3.0"
```
Read more [here](https://aquasecurity.github.io/trivy/v0.53/docs/coverage/language/dart/#sdk-dependencies).

## ☕  Support for SBT Projects with sbt-dependency-lock 🔒 
Trivy now supports scanning Scala projects using sbt with lockfiles generated by [sbt-dependency-lock](https://stringbean.github.io/sbt-dependency-lock/).

Read more [here](https://aquasecurity.github.io/trivy/v0.53/docs/coverage/language/java/#sbt).

Thanks to @stringbean

## 🗃️ Support for `maven-metadata.xml` for `maven snapshot repositories` 📥
Some `maven snapshot repositories` (e.g. `oss.sonatype.org`) use `maven-metadata.xml` with the latest pom file name.
Trivy now parses this file. 
If the file doesn't exist or doesn't contain pom file name information - Trivy checks the `<artifactId>-<version>.pom` file (as for `release` repositories).

## 🎊 Support for `installed.json` PHP files ✨
Trivy now scans `installed.json` files (available in `image` and `rootfs` modes).

```bash
installed.json (composer-installed)

Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 2, HIGH: 0, CRITICAL: 0)

┌─────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬────────────────────────────────────────────────────┐
│     Library     │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                       Title                        │
├─────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼────────────────────────────────────────────────────┤
│ guzzlehttp/psr7 │ CVE-2022-24775 │ MEDIUM   │ fixed  │ 1.8.3             │ 1.8.4, 2.1.1  │ Improper Input Validation in guzzlehttp/psr7       │
│                 │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2022-24775         │
│                 ├────────────────┤          │        │                   ├───────────────┼────────────────────────────────────────────────────┤
│                 │ CVE-2023-29197 │          │        │                   │ 1.9.1, 2.4.5  │ Improper header name validation in guzzlehttp/psr7 │
│                 │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2023-29197         │
└─────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴────────────────────────────────────────────────────┘
```
Read more [here](https://aquasecurity.github.io/trivy/v0.53docs/coverage/language/php/#installedjson).

## 🌀 Trivy migrated to `CycloneDX v1.6` 🌪️
Trivy now uses [CycloneDX v1.6](https://cyclonedx.org/news/cyclonedx-v1.6-released/) for `cyclonedx` format.

## 🧠 Memory Cache Backend ⚡
Trivy introduces a memory cache backend, enhancing scan efficiency for repositories, filesystems, and SBOMs by reducing unnecessary disk I/O operations. By default, memory backend is used for these scan targets, so no changes are required. For container images and VM images, the filesystem cache remains in use, but you can specify `--cache-backend memory` if you don't need a cache for subsequent scans.

```bash
$ trivy image debian:11 --cache-backend memory
```

### 🥡 Node collector commands ⛏️

The "node-collector" component, that collects data from within nodes for evaluation by checks, is now more flexible and extensible. The Trivy compliance definision now has a new `commands` field that refers to scriptlets that collect data by the node-collector. The data is then available in rego checks for evaluation. This makes it easier for users to contribute and maintain specs.

For example, here's a snippet from the new CIS benchmark for Kubernetes:

```yaml
spec:
  id: k8s-cis-1.23
  title: CIS Kubernetes Benchmarks v1.23
  description: CIS Kubernetes Benchmarks
  platform: k8s
  type: cis
  version: '1.23'
  relatedResources:
  - https://www.cisecurity.org/benchmark/kubernetes
  controls:
  - id: 1.1.1
    name: Ensure that the API server pod specification file permissions are set to
      600 or more restrictive
    description: Ensure that the API server pod specification file has permissions
      of 600 or more restrictive
    checks:
    - id: AVD-KCV-0073
    commands:
    - id: CMD-0001
    severity: HIGH
```

for more details check our docs on [compliance contribution](https://github.com/aquasecurity/trivy/blob/main/docs/docs/compliance/compliance.md#contribute-a-built-in-compliance-report)

### 🧾 New compliance specs ☑️

- eks-cis-1.4
- rke2-cis-1.24

We are working on adding more platforms. Please let us know if yould like to help!

## 🫔 Selector support for all providers 🍏

Trivy now supports selectors and subtypes for all cloud providers.

```rego
# METADATA
# title: "Test"
# scope: package
# schemas:
# - input: schema["cloud"]
# custom:
#   id: TEST001
#   avd_id: TEST001
#   severity: LOW
#   input:
#     selector:
#     - type: cloud
#       subtypes:
#         - service: compute
#           provider: google
``` 

Previously Trivy only supported AWS and Kubernetes.

## 🐳 New resources support in misconfiguration scanning 🎍

We have added the following resources as part of our misconfiguration scanning to enhance our scanning posture.

- [AWS EC2 SecurityGroupIngress](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-securitygroupingress.html)
- [AWS EC2 SecurityGroupEgress](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-securitygroupegress.html)
- [AWS API Gateway](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_rest_api)

## 📦 Plugin GitHub Archives Support 🗂️
This update introduces support for nested archives in Trivy plugins. Users can now directly download and install plugins from GitHub repository archives, simplifying the installation process.

```
$ trivy plugin install https://github.com/aquasecurity/trivy-plugin-referrer/archive/refs/heads/main.zip
```

# 👷‍♂️ Notable Fixes 🛠️
- https://github.com/aquasecurity/trivy/issues/7006
- https://github.com/aquasecurity/trivy/issues/6941
- https://github.com/aquasecurity/trivy/issues/6931
- https://github.com/aquasecurity/trivy/issues/6946
- https://github.com/aquasecurity/trivy/issues/6801
- https://github.com/aquasecurity/trivy/issues/6915
- https://github.com/aquasecurity/trivy/issues/6981
- https://github.com/aquasecurity/trivy/issues/7007
- https://github.com/aquasecurity/trivy/issues/7050
- https://github.com/aquasecurity/trivy/issues/6865
- https://github.com/aquasecurity/trivy/issues/6817
- https://github.com/aquasecurity/trivy/issues/6848
- https://github.com/aquasecurity/trivy/issues/6854

**Comment by stringbean** [comment]
This release also adds support for scanning sbt projects (Scala/Java) that use sbt-dependency-lock: java: add support for sbt projects using sbt-dependency-lock (#6882).

**Reply by knqyf263** [reply]
Oooops, we're so sorry to miss it. We'll add it.

**Reply by knqyf263** [reply]
Added
